### PR TITLE
Route MoE execution through kernels runtime API

### PR DESCRIPTION
**Pull request by vikhyat**
## Summary

Routes `FusedMoEModule` through `kestrel_kernels.moe.prepare/pack_weights/forward` instead of managing padded routing buffers, fused MoE workspaces, base GEMM dispatch, activation, sum, and LoRA application directly in the engine.

This removes the old engine-side direct MoE path (`kestrel/fused_moe/_direct.py`), removes the shared base-MoE workspace preallocation API, and deletes the stale prepared-forward orchestration from `FusedMoEModule`. The existing CUDA graph warmup now prepares `MoeHandle` workspaces before graph capture; Kestrel only passes semantic `prefill`/`decode` mode and the kernels runtime owns backend/variant selection.

Depends on the kernels PR that adds `kestrel_kernels.moe`.

## Validation

- `python -m py_compile kestrel/fused_moe/module.py kestrel/fused_moe/__init__.py kestrel/moondream/runtime.py` -> ok
- Full ChartQA B200 runs against this branch showed accuracy within legacy run-to-run variance
- B200 graph-captured FP8 MoE microbench showed parity with the legacy path for decode buckets 1, 2, 4, 8, 16, 32, 48, and 64

### CODE REVIEW

**Comment by vikhyat** [comment]
@codex review

File: kestrel/scheduler/scheduler.py [hunk 1]
```diff
@@ -1350,8 +1380,33 @@ def _sample_batch(
...
+    def _selected_logprobs_from_logits(
+        logits: Tensor,
+        sampled_ids: Tensor,
+        temperatures: Tensor,
+    ) -> Tensor:
+        temps = torch.clamp(temperatures, min=0.0)
+        effective_temp = torch.where(
+            temps > _SAMPLING_EPS,
+            temps,
+            torch.ones_like(temps),
```
[the remark this answers is not in the document]
Comment (vikhyat) [reply kestrel/scheduler/scheduler.py:1404]: I do not think the comment is correct as stated: `sample_step_from_logits` defines `temperature <= _SAMPLING_EPS` as force-greedy, not as a low-temperature softmax. I did fix the related recompute mismatch in 4fa4e2b: suppressed-baseline logprobs now mirror sampler semantics, returning greedy logprobs for `<= eps`/invalid rows and preserving temperature-scaled logprobs only for `> eps`. Added coverage for the epsilon boundary.

File: kestrel/moondream/moe.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+            max_loras=max_loras,
+            mode=mode,
+        )
+        key = (
+            str(hidden_states.device),
+            hidden_states.dtype,
+            weight_format,
+            capacity.max_tokens,
+            capacity.max_loras,
+            capacity.mode,
```
[the remark this answers is not in the document]
Comment (vikhyat) [reply kestrel/moondream/moe.py:148]: Fixed in 4fa4e2b: prefill MoE handles now use power-of-two capacity buckets and reuse the cached handle within that bucket. Decode handles still use exact token capacity so CUDA graph bucket captures keep fixed shapes.

**Comment by vikhyat** [comment]
@codex review

**Comment by vikhyat** [comment]
@codex review

**Comment by vikhyat** [comment]
@codex review
